### PR TITLE
Housekeeping - Nov'23

### PR DESCRIPTION
**Pull request by prasanna214**
## Proposed changes

1. Upgraded the following to their latest versions
    - Springboot
    - Gradle
    - Axon server SE
    - KeyCloak
2. Removed deprecated methods of spring security package in `WebSecurityConfig`
3. Updated ReadMe with instruction about notebook usage
4. Included `docker compose up` in the gradle `bootRun` task

## Checklist

- [x] My local build passed (You ran `./gradlew clean build` without failures)
- [x] I have added necessary documentation (if appropriate)
- [x] I have assigned a label to the PR 
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/api/src/main/java/engineering/everest/lhotse/api/config/WebSecurityConfig.java [hunk 1]
```diff
@@ -29,23 +32,24 @@ public SecurityFilterChain configure(HttpSecurity http, KeycloakJwtGrantedAuthor
...
-            .requestMatchers("/api/**").authenticated()
-            .requestMatchers("/actuator/health/**", "/actuator/metrics/**", "/actuator/prometheus").hasAnyRole("ADMIN", "MONITORING")
-            .requestMatchers("/actuator/**", "/admin/**").hasRole("ADMIN")
-            .anyRequest().permitAll()
-            .and().logout().logoutRequestMatcher(new AntPathRequestMatcher("/logout", "GET")).logoutSuccessUrl("/")
-            .and().oauth2ResourceServer(configurer -> configurer.jwt().jwtAuthenticationConverter(jwtAuthenticationConverter));
+        http.cors(withDefaults()).csrf(AbstractHttpConfigurer::disable)
+            .authorizeHttpRequests(authz -> authz
+                .requestMatchers(
+                    "/api/organizations/**",
```
Comment (sluehr) [inline src/api/src/main/java/engineering/everest/lhotse/api/config/WebSecurityConfig.java]: We don't have orgs any more :)

File: docker-compose.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ services:
       POSTGRES_PASSWORD: postgres
 
   axonserver:
-    image: axoniq/axonserver:4.6.10-dev
+    image: axoniq/axonserver:4.6.11-dev
```
Comment (sluehr) [inline docker-compose.yml]: Can you check what the latest is? The image tags have changed to something like `2023.2.1-jdk-17-nonroot`

**Review by sluehr — APPROVED** [review]
LGTM but please check comments

File: docker-compose.yml (hunk 2 above)
Comment (prasanna214) [reply docker-compose.yml]: You are right!
`4.6.11` was the latest version for SE only image.  But they have combined both EE and SE into one image after that release.  Upgraded ours to the latest one now ✅

Comment (prasanna214) [reply docker-compose.yml]: Removed ✅

File: src/api/src/main/java/engineering/everest/lhotse/api/config/WebSecurityConfig.java (hunk 1 above)
Comment (prasanna214) [reply src/api/src/main/java/engineering/everest/lhotse/api/config/WebSecurityConfig.java]: Removed ✅

**Pull request merged**